backend/models/project.js

```diff
@@ -18,266 +18,204 @@
+		await checkDupName(teamspace, project);
+		await checkPermissionName(project.permissions);
 
-			const userPromises = [];
+		await db.insert(teamspace, PROJECTS_COLLECTION_NAME, project);
 
-			usersToRemove.forEach(user => {
-				// remove all model permissions in this project as well, if any
-				userPromises.push(
-					ModelSetting.find(this._dbcolOptions, { "permissions.user": user}).then(settings =>
-						Promise.all(
-							settings.map(s => s.changePermissions(s.permissions.filter(perm => perm.user !== user)))
-						)
-					)
-				);
-			});
+		project.permissions = C.IMPLIED_PERM[C.PERM_PROJECT_ADMIN].project;
 
-			return Promise.all(userPromises);
-		}).then(() => {
-			return this.save();
-		});
+		return project;
 	};
 
-	schema.statics.findAndPopulateUsers = function(account, query) {
-		const User = require("./user");
-		let userList;
-
-		return User.getAllUsersInTeamspace(account.account).then(users => {
-			userList = users;
-			return Project.find(account, query);
-		}).then(projects => {
-			if (projects) {
-				projects.forEach(p => Project.populateUsers(userList, p));
-			}
+	Project.delete = async function(teamspace, name) {
+		const ModelHelper = require("./helper/model");
+		const project = await Project.findOneProject(teamspace, {name});
 
-			return projects;
-		});
+		if (!project) {
+			throw responseCodes.PROJECT_NOT_FOUND;
+		}
+
+		await db.remove(teamspace, PROJECTS_COLLECTION_NAME, {name});
+		// remove all models as well
+		if (project.models) {
+			await Promise.all(project.models.map(m => ModelHelper.removeModel(teamspace, m, true)));
+		}
+
+		return project;
 	};
 
-	schema.statics.findOneAndPopulateUsers = function(account, query) {
+	Project.listProjects = async function(teamspace, query = {}, projection) {
 		const User = require("./user");
-		let userList;
-
-		return User.getAllUsersInTeamspace(account.account).then(users => {
-			userList = users;
-			return Project.findOne(account, query);
-		}).then(project => {
-			if (project) {
-				return Project.populateUsers(userList, project);
-			} else {
-				return Promise.reject(responseCodes.PROJECT_NOT_FOUND);
-			}
-		});
-	};
+		const userList = await User.getAllUsersInTeamspace(teamspace);
+		const projects = await db.find(teamspace, PROJECTS_COLLECTION_NAME, query, projection);
 
-	schema.statics.findByNames = function(account, projectNames) {
-		return Project.find({account}, { name: { $in:projectNames } });
-	};
+		if (projects) {
+			projects.forEach(p => {
+				prepareProject(p);
+				populateUsers(userList, p);
+			});
+		}
 
-	schema.statics.findByIds = function(account, _ids) {
-		return Project.find({account}, { _id: { $in: _ids.map(utils.stringToUUID) } });
+		return projects;
 	};
 
-	schema.statics.populateUsers = function(userList, project) {
-		userList.forEach(user => {
-			const userFound = project.permissions.find(perm => perm.user === user);
+	Project.findProjectAndPopulateUsers = async function(teamspace, projectName) {
```

Comment:
This needs better naming. The naming is really misleading.

After looking around, this is only used by the endpoint, which looks like it's to show the permissions page on the frontend. So it needs to know all the users within teamspace and whether they have any permissions.

PopulateUsers suggests we're doing a write. This is more like `getAllUsersProjectPermissions`  or something (i'm bad at naming, but hopefully you see the direction i'm trying to go towards)